homeassistant/components/nest/local_auth.py

```diff
@@ -0,0 +1,44 @@
+"""Local Nest authentication."""
+from functools import partial
+
+from aiohttp.client_exceptions import ClientError
+
+from homeassistant.core import callback
+from . import config_flow
+from .const import DOMAIN
+
+ACCESS_TOKEN_URL = 'https://api.home.nest.com/oauth2/access_token'
```

Comment:
Should we get those URLs from python-nest?
https://github.com/jkoelker/python-nest/blob/a45cda8c4efae0ec0725fe15d2788928cd54f21d/nest/nest.py#L22-L23

Reply:
Good point. I've rewritten to use this whole file to use the authentication mechanism from `python-nest`. I've added a test so that if the package changes, it will blow up our test.